integration_tests/src/main/python/parquet_write_test.py

```diff
@@ -32,16 +33,30 @@
 writer_confs={'spark.sql.legacy.parquet.datetimeRebaseModeInWrite': 'CORRECTED',
               'spark.sql.legacy.parquet.int96RebaseModeInWrite': 'CORRECTED'}
 
+# https://github.com/rapidsai/cudf/issues/7152
+# The above bug in cudf causes problem writing Decimals with precision < 10
+random.seed(23423)
 parquet_write_gens_list = [
-        [byte_gen, short_gen, int_gen, long_gen, float_gen, double_gen,
-            string_gen, boolean_gen, date_gen, timestamp_gen]]
+    [byte_gen, short_gen, int_gen, long_gen, float_gen, double_gen,
+     string_gen, boolean_gen, date_gen, timestamp_gen,
+     DecimalGen(precision=10, scale=random.randint(1,10)),
+     DecimalGen(precision=11, scale=random.randint(1,11)),
+     DecimalGen(precision=12, scale=random.randint(1,12)),
+     DecimalGen(precision=13, scale=random.randint(1,13)),
+     DecimalGen(precision=14, scale=random.randint(1,14)),
+     DecimalGen(precision=15, scale=random.randint(1,15)),
+     DecimalGen(precision=16, scale=random.randint(1,16)),
+     DecimalGen(precision=17, scale=random.randint(1,17)),
+     DecimalGen(precision=18, scale=random.randint(1,18))]
+]
 
 parquet_ts_write_options = ['INT96', 'TIMESTAMP_MICROS', 'TIMESTAMP_MILLIS']
 
 @pytest.mark.parametrize('parquet_gens', parquet_write_gens_list, ids=idfn)
 @pytest.mark.parametrize('reader_confs', reader_opt_confs)
 @pytest.mark.parametrize('v1_enabled_list', ["", "parquet"])
 @pytest.mark.parametrize('ts_type', parquet_ts_write_options)
+@allow_non_gpu("CoalesceExec")
```

Comment:
why is this?

Reply:
Because `CoalesceExec` doesn't support `Decimals` yet

Reply:
Do we want to rather write a separate test for Decimals so we are at least testing coalesce with other types? we already have other tests testing coalesce though

Reply:
yes, I don't want it to be ok if other types have that